x-pack/legacy/plugins/triggers_actions_ui/np_ready/public/application/components/builtin_alert_types/threshold/visualization.tsx

```diff
@@ -0,0 +1,304 @@
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Fragment, useEffect, useState } from 'react';
+import { IUiSettingsClient } from 'kibana/public';
+import { i18n } from '@kbn/i18n';
+import {
+  AnnotationDomainTypes,
+  Axis,
+  getAnnotationId,
+  getAxisId,
+  getSpecId,
+  Chart,
+  LineAnnotation,
+  LineSeries,
+  Position,
+  ScaleType,
+  Settings,
+} from '@elastic/charts';
+import { TimeBuckets } from 'ui/time_buckets';
+import dateMath from '@elastic/datemath';
+import moment from 'moment-timezone';
+import { EuiCallOut, EuiLoadingChart, EuiSpacer, EuiEmptyPrompt, EuiText } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { npStart } from 'ui/new_platform';
+import { getThresholdAlertVisualizationData } from './lib/api';
+import { comparators, aggregationTypes } from './expression';
+import { useAppDependencies } from '../../../app_dependencies';
+import { SectionError } from '../../../components/page_error/section_error';
+import { Alert } from '../../../../types';
+
+const customTheme = () => {
+  return {
+    lineSeriesStyle: {
+      line: {
+        strokeWidth: 3,
+      },
+      point: {
+        visible: false,
+      },
+    },
+  };
+};
+
+const getTimezone = (uiSettings: IUiSettingsClient) => {
+  const config = uiSettings;
+  const DATE_FORMAT_CONFIG_KEY = 'dateFormat:tz';
+  const isCustomTimezone = !config.isDefault(DATE_FORMAT_CONFIG_KEY);
+  if (isCustomTimezone) {
+    return config.get(DATE_FORMAT_CONFIG_KEY);
+  }
+
+  const detectedTimezone = moment.tz.guess();
+  if (detectedTimezone) {
+    return detectedTimezone;
+  }
+  // default to UTC if we can't figure out the timezone
+  const tzOffset = moment().format('Z');
+  return tzOffset;
+};
+
+const getDomain = (alertParams: any) => {
+  const VISUALIZE_TIME_WINDOW_MULTIPLIER = 5;
+  const fromExpression = `now-${alertParams.timeWindowSize * VISUALIZE_TIME_WINDOW_MULTIPLIER}${
+    alertParams.timeWindowUnit
+  }`;
+  const toExpression = 'now';
+  const fromMoment = dateMath.parse(fromExpression);
+  const toMoment = dateMath.parse(toExpression);
+  const visualizeTimeWindowFrom = fromMoment ? fromMoment.valueOf() : 0;
+  const visualizeTimeWindowTo = toMoment ? toMoment.valueOf() : 0;
+  return {
+    min: visualizeTimeWindowFrom,
+    max: visualizeTimeWindowTo,
+  };
+};
+
+const getThreshold = (alertParams: any) => {
+  return alertParams.threshold.slice(
+    0,
+    comparators[alertParams.thresholdComparator].requiredValues
+  );
+};
+
+const getTimeBuckets = (alertParams: any) => {
+  const domain = getDomain(alertParams);
+  const timeBuckets = new TimeBuckets();
+  timeBuckets.setBounds(domain);
+  return timeBuckets;
+};
+
+interface Props {
+  alert: Alert;
+}
+
+export const ThresholdVisualization: React.FunctionComponent<Props> = ({ alert }) => {
```

Comment:
Same question about tests for this one... I'd be worried maintaining this code without tests, as I don't know what the expected behaviour or intent is 🤷‍♂️

Reply:
because not a part of 7.6 and still in progress :)